### PR TITLE
Add ct_stat constants and related usage test #121

### PR DESCRIPTION
**Pull request by lyonsdpy**
resolve: #121

### CODE REVIEW

**Review by stapelberg — COMMENTED** [review]
Can you give me a link to the Linux kernel header file where you got these constant values from? I want to see if we can add them to https://pkg.go.dev/golang.org/x/sys/unix instead of hard-coding their values here.

**Comment by lyonsdpy** [comment]
> Can you give me a link to the Linux kernel header file where you got these constant values from? I want to see if we can add them to https://pkg.go.dev/golang.org/x/sys/unix instead of hard-coding their values here.

- First I find the values from [nftables-datatye](https://wiki.nftables.org/wiki-nftables/index.php/Data_types#Conntrack_types)

- After saw your comments, I got these values defined in `/usr/include/linux/netfilter/nf_conntrack_common.h`
```C
enum ip_conntrack_info {
	/* Part of an established connection (either direction). */
	IP_CT_ESTABLISHED,

	/* Like NEW, but related to an existing connection, or ICMP error
	   (in either direction). */
	IP_CT_RELATED,

	/* Started a new connection to track (only
           IP_CT_DIR_ORIGINAL); may be a retransmission. */
	IP_CT_NEW,

	/* >= this indicates reply direction */
	IP_CT_IS_REPLY,

	IP_CT_ESTABLISHED_REPLY = IP_CT_ESTABLISHED + IP_CT_IS_REPLY,
	IP_CT_RELATED_REPLY = IP_CT_RELATED + IP_CT_IS_REPLY,
	/* No NEW in reply direction. */

	/* Number of distinct IP_CT types. */
	IP_CT_NUMBER,

	/* only for userspace compatibility */
	IP_CT_NEW_REPLY = IP_CT_NUMBER,
};

#define NF_CT_STATE_INVALID_BIT			(1 << 0)
#define NF_CT_STATE_BIT(ctinfo)			(1 << ((ctinfo) % IP_CT_IS_REPLY + 1))
#define NF_CT_STATE_UNTRACKED_BIT		(1 << 6)
```

- An example
```cpp
static const struct symbol_table ct_state_tbl = {
	.symbols	= {
		SYMBOL("invalid",	NF_CT_STATE_INVALID_BIT),   // == 1
		SYMBOL("new",		NF_CT_STATE_BIT(IP_CT_NEW)),  // ==2
		SYMBOL("established",	NF_CT_STATE_BIT(IP_CT_ESTABLISHED)), // ==4
		SYMBOL("related",	NF_CT_STATE_BIT(IP_CT_RELATED)),  // ==8
		SYMBOL("untracked",	NF_CT_STATE_UNTRACKED_BIT), // ==64
		SYMBOL_LIST_END
	}
};
```

**Review by stapelberg — CHANGES_REQUESTED** [review]

File: expr/ct.go [hunk 1]
```diff
@@ -47,6 +47,12 @@ const (
 	CtKeyAVGPKT     CtKey = unix.NFT_CT_AVGPKT
 	CtKeyZONE       CtKey = unix.NFT_CT_ZONE
 	CtKeyEVENTMASK  CtKey = unix.NFT_CT_EVENTMASK
+
+	CtStatINVALID     uint32 = 1
```
[the remark this answers is not in the document]
Comment (lyonsdpy) [reply expr/ct.go:51]: Ok,I'll change this later.

File: expr/ct.go [hunk 2]
```diff
@@ -47,6 +47,13 @@ const (
 	CtKeyAVGPKT     CtKey = unix.NFT_CT_AVGPKT
 	CtKeyZONE       CtKey = unix.NFT_CT_ZONE
 	CtKeyEVENTMASK  CtKey = unix.NFT_CT_EVENTMASK
+
+	// https://sources.debian.org/src//nftables/0.9.8-3/src/ct.c/?hl=39#L39
+	CtStatBitINVALID     uint32 = 1
```
Comment (stapelberg) [inline expr/ct.go:52]: This still reads CtStatBit, but should read CtStateBit (note the `e`)

Comment (lyonsdpy) [reply expr/ct.go]: sorry about that. I now added the `e`

**Review by stapelberg — APPROVED** [review]

**Pull request merged**